modules/parquet/src/parquetjs/classes/parquet-envelope-reader.ts

```diff
@@ -0,0 +1,128 @@
+// Forked from https://github.com/kbajalc/parquets under MIT license (Copyright (c) 2017 ironSource Ltd.)
+import {ParquetSchema} from '../schema/schema';
+import {PARQUET_MAGIC, PARQUET_MAGIC_ENCRYPTED} from '../../constants';
+import {ColumnChunk, CompressionCodec, FileMetaData, RowGroup, Type} from '../parquet-thrift';
+import {ParquetBuffer, ParquetCompression, ParquetData, PrimitiveType} from '../schema/declare';
+import {decodeDataPages} from '../utils/decode-utils';
+import {fstat, fopen, fread, fclose} from '../utils/file-utils';
+import {decodeFileMetadata, getThriftEnum, fieldIndexOf} from '../utils/read-utils';
+
+/**
+ * The parquet envelope reader allows direct, unbuffered access to the individual
+ * sections of the parquet file, namely the header, footer and the row groups.
+ * This class is intended for advanced/internal users; if you just want to retrieve
+ * rows from a parquet file use the ParquetReader instead
+ */
+export class ParquetEnvelopeReader {
+  public read: (position: number, length: number) => Promise<Buffer>;
+  /**
+   * Close this parquet reader. You MUST call this method once you're finished
+   * reading rows
+   */
+  public close: () => Promise<void>;
+  public fileSize: number;
+
+  static async openFile(filePath: string): Promise<ParquetEnvelopeReader> {
+    const fileStat = await fstat(filePath);
+    const fileDescriptor = await fopen(filePath);
+
+    const readFn = fread.bind(undefined, fileDescriptor);
+    const closeFn = fclose.bind(undefined, fileDescriptor);
+
+    return new ParquetEnvelopeReader(readFn, closeFn, fileStat.size);
+  }
+
+  static async openBuffer(buffer: Buffer): Promise<ParquetEnvelopeReader> {
+    const readFn = (position: number, length: number) =>
+      Promise.resolve(buffer.slice(position, position + length));
+    const closeFn = () => Promise.resolve();
+    return new ParquetEnvelopeReader(readFn, closeFn, buffer.length);
+  }
```

Comment:
Nit For separate PR: we should be able to drop these. We are using the constructor directly now in the Reader for several cases, no need to have extra functions here.

Less code in fewer places will make the code base easier to maintain.